### PR TITLE
common: resolve recursively include in root manifest

### PR DESCRIPTION
**Pull request by massonju**
The root manifest may have 'include' tag, we should resolve recursively these elements and add the contents of the subsequent manifests found in the root manifest.

### CODE REVIEW

**Comment by massonju** [comment]
This PR depends on https://github.com/makohoek/demo-manifests/pull/5

File: repo_resource/test_check.py [hunk 1]
```diff
@@ -78,6 +78,21 @@ def setUp(self):
...
             }
         }
+        self.include_one_project_source = {
+            'source': {
+                'url': 'https://github.com/makohoek/demo-manifests.git',
+                'revision': 'main',
+                'name': 'aosp_include_one_project.xml'
+            }
+        }
+
```
Comment (makohoek) [inline repo_resource/test_check.py:88]: nit: drop blank line please (to do the same as the other members)

File: repo_resource/common.py [hunk 2]
```diff
@@ -376,6 +376,24 @@ def __manifest_out(self, filename):
         finally:
             self.__restore_oldpwd()
 
+    def resolve_manifest_include(self, manifest):
+        for idx, element in enumerate(list(manifest.iter())):
+            if element.tag == 'include':
+                sub_manifest_name = element.get('name')
+                sub_xml = ET.parse('.repo/manifests/'+sub_manifest_name)
```
Comment (makohoek) [inline repo_resource/common.py:383]: What happens when the include points towards a file that does not exist?

How does the error look like?

If the error message is unclear, please add some error handling in here to make sure that the caller has a proper error message

**Review by makohoek — COMMENTED** [review]
This looks great already. I have one nitpick and one question. Please have a look.

File: repo_resource/common.py (hunk 2 above)
Comment (massonju) [reply repo_resource/common.py:383]: By default that will return this kind of errors:
```
fatal: manifest 'aosp_include_invalid_project.xml' not available
fatal: include [/tmp/repo-resource-cache/.repo/manifests/]aosp_invalid.xml doesn't exist or isn't a file
```

Is it necessary for you ?

File: repo_resource/test_check.py (hunk 1 above)
Comment (massonju) [reply repo_resource/test_check.py:88]: yes I will change it next version

**Review by makohoek — APPROVED** [review]

**Comment by massonju** [comment]
@makohoek thanks

**Pull request merged**